test/integration/pipeline/tx_pipeline_integration_test.cpp

```diff
@@ -219,3 +236,89 @@ TEST_F(TxPipelineIntegrationTest, TxPipelineTest) {
 
   validate();
 }
+
+/**
+ * @given Admin creates domains: [usabnk, ru, khm]
+ *        National Bank of America, creates domain usabnk (nba@usabnk) and asset usd,
+ * @when A user ivan in domain of Russia (ivan@ru) wants to possess and
+ *       transfer asset usd#usabnk to user tea in domain of Cambodia (tea@khm).
+ * @then Validate transactions processed.
+ */
+TEST_F(TxPipelineIntegrationTest, TransferAssetTestAmongDomainTest) {
+  const std::string DOMAIN_USABANK = "usabnk";
+  const std::string DOMAIN_RU = "ru";
+  const std::string DOMAIN_KHM = "khm";
+  const std::string USER_DEFAULT_ROLE = "user";
+  const std::string MONEY_CREATOR_ROLE = "money_creator";
+  const std::string ASSET_USD_NAME = "usd";
+  const std::string ASSET_USD_ID = ASSET_USD_NAME + "#" + DOMAIN_USABANK;
+  const std::string NBA_NAME = "nba";
+  const std::string IVAN_NAME = "ivan";
+  const std::string TEA_NAME = "tea";
+  const std::string ADMIN_ID = "admin@test";
+  const std::string NBA_ID = NBA_NAME + "@" + DOMAIN_USABANK;
+  const std::string IVAN_ID = IVAN_NAME + "@" + DOMAIN_RU;
+  const std::string TEA_ID = TEA_NAME + "@" + DOMAIN_KHM;
+
+  auto adminKeypair = iroha::KeysManagerImpl(ADMIN_ID).loadKeys().value();
+  auto nbaKeypair = createNewAccountKeypair(NBA_ID, "nba_pass").value();
+  auto ivanKeypair = createNewAccountKeypair(IVAN_ID, "ivan_pass").value();
+  auto teaKeypair = createNewAccountKeypair(TEA_ID, "tea_pass").value();
+
+  // Admin creates domains: [usabnk, khm, ru]
+  auto tx1 = TransactionGenerator().generateTransaction(
+    ADMIN_ID, 1, {
+      CommandGenerator().generateCreateDomain(
+        DOMAIN_USABANK, MONEY_CREATOR_ROLE),
+      CommandGenerator().generateCreateDomain(
+        DOMAIN_RU, USER_DEFAULT_ROLE),
+      CommandGenerator().generateCreateDomain(
+        DOMAIN_KHM, USER_DEFAULT_ROLE),
+      CommandGenerator().generateCreateAccount(
+        NBA_NAME, DOMAIN_USABANK, nbaKeypair.pubkey),
+      CommandGenerator().generateCreateAccount(
+        IVAN_NAME, DOMAIN_RU, ivanKeypair.pubkey),
+      CommandGenerator().generateCreateAccount(
+        TEA_NAME, DOMAIN_KHM, teaKeypair.pubkey)
+    });
+
+  iroha::model::ModelCryptoProviderImpl(adminKeypair).sign(tx1);
+
+  // creates default domains and usd asset. add asset qty to nba
+  auto tx2 = TransactionGenerator().generateTransaction(
+    NBA_ID, 1, {
+      CommandGenerator().generateCreateAsset(
```

Comment:
Extract all of these generated `*Asset*` commands, that also significantly improve readabilty

Reply:
I think passing this transaction into consensus is better because of removing transactions from genesis block as possible.